default.nix

```diff
@@ -1,14 +1,40 @@
 {
   niv ? import nix/sources.nix,
-  pkgs ? import niv.nixpkgs { config = { allowUnfree = true; }; }
+  nixpkgs ? niv.nixpkgs,
+  pkgs ? import nixpkgs {}
 }:
+
 let
-  cbsLib = import ./lib { inherit pkgs; };
-  cbspkgs = import ./pkgs { inherit niv pkgs cbsLib; };
-  cbsModules = import ./modules { inherit niv cbspkgs pkgs; };
-in cbspkgs //
-  {
-    lib = cbsLib;
-    nixpkgs = pkgs;
-    modules = cbsModules;
-  }
+  lib = import ./lib { inherit pkgs; };
+  initrdSet = import ./pkgs/initrd-creator/release.nix {
+    inherit pkgs;
+    cbsLib = lib;
+  };
+  kernelSet = import ./pkgs/kernels { inherit pkgs; };
+in pkgs.recurseIntoAttrs rec {
+  inherit (niv) nixpkgs;
+  inherit lib;
+  modules = import ./modules { inherit niv; };
+
+  bender = pkgs.callPackage niv.bender {};
+
+  hydra = pkgs.callPackage ./pkgs/hydra { src = niv.hydra; };
+
+  initrds = pkgs.recurseIntoAttrs
+    (builtins.mapAttrs (_: pkgs.recurseIntoAttrs) initrdSet);
+
+  ipxe = pkgs.callPackage ./pkgs/ipxe { src = niv.ipxe; };
+
+  run-sotest = pkgs.callPackage ./pkgs/run-sotest {};
+
+  sotest-kernels = pkgs.recurseIntoAttrs kernelSet;
+
+  sotest-testruns = pkgs.recurseIntoAttrs (import ./pkgs/sotest-testruns {
+    inherit pkgs;
+    inherit (initrdSet) initrds;
+    cbsLib = lib;
+    kernels = kernelSet;
+  });
```

Comment:
it's a bit ugly that we can't just use `sotest-kernels` and `initrds` here because of `recurseForDerivations`. Would it make sense to handle `recurseForDerivations` as a special case in `./pkgs/sotest-testruns`?